packages/cloud-functions/src/notifications/index.ts

```diff
@@ -0,0 +1,96 @@
+import * as admin from 'firebase-admin'
+import i18next, { TFunction } from 'i18next'
+import { NOTIFICATIONS_TTL_MS } from '../config'
+import { database, messaging } from '../firebase'
+
+const TAG = 'Notifications'
+
+export async function getTranslatorForAddress(address: string): Promise<TFunction> {
+  return new Promise((resolve, reject) => {
+    database()
+      .ref(`registrations/${address}/language`)
+      .once('value', (snapshot) => {
+        const language = snapshot.val()
+        if (language) {
+          resolve(i18next.getFixedT(language))
+        } else {
+          resolve(i18next.t.bind(i18next))
+        }
+      })
+      .catch(reject)
+  })
+}
+
+export async function getTokenFromAddress(address: string): Promise<string | null> {
+  return new Promise((resolve, reject) => {
+    database()
+      .ref(`registrations/${address}/fcmToken`)
+      .once('value', (snapshot) => {
+        const token = snapshot.val()
+        resolve(token)
+      })
+      .catch(reject)
+  })
+}
+
+export async function sendNotification(
+  title: string,
+  body: string,
+  address: string,
+  data: { [key: string]: string }
+) {
+  const token = await getTokenFromAddress(address)
+  if (!token) {
+    console.info(TAG, 'FCM token missing for address:', address)
+    return
+  }
+
+  // https://firebase.google.com/docs/cloud-messaging/concept-options#setting-the-priority-of-a-message
+  const message: admin.messaging.Message = {
+    notification: {
+      title,
+      body,
+    },
+    android: {
+      ttl: NOTIFICATIONS_TTL_MS,
+      priority: 'normal',
+      notification: {
+        icon: 'ic_stat_rings',
+        color: '#42D689',
+      },
+    },
+    data,
+    token,
+  }
+
+  try {
+    // TODO: Use a logger that abstracts the json stringifying thing.
```

Comment:
`firebase-functions` provides this out of the box: https://github.com/firebase/firebase-functions/issues/612#issuecomment-771762956

Reply:
I saw that library, but I wanted to use something that we could also use in the `blockchain-api` and the `notification-service` which is why I didn't use it. I will come back to this after I've made some progress on the cEUR stuff that I've left behind the last couple of weeks.